app/lib/utils/queryMapper.js

```diff
@@ -1,26 +1,67 @@
 const constants = require('../constants');
-const comparisons = require('./comparisons');
+const utils = require('./utils');
+
+function isProfessionalChoice(query) {
+  return (utils.areEqual(query.type, constants.serviceTypes.professional)
+    && (utils.getValues(constants.serviceChoices).includes(query.origin)));
+}
 
 function getLocationHeading(query) {
   if (query.type) {
-    if (comparisons.areEqual(query.type, constants.serviceTypes.professional)
-      && (comparisons.getValues(constants.serviceChoices).includes(query.origin))) {
+    if (isProfessionalChoice(query)) {
       return 'Where would you like to see a sexual health professional?';
     }
-    if (comparisons.areEqual(query.type, constants.serviceTypes.kit)) {
+    if (utils.areEqual(query.type, constants.serviceTypes.kit)) {
       if (query.origin === constants.serviceChoices['16to25']) {
         return 'Where would you like to collect your free test kit?';
       } else if (query.origin === constants.serviceChoices.over25) {
-        return 'Where would you like to collect your test kit?';
+        return 'Where would you like to buy your test kit?';
       }
     }
-    if (comparisons.areEqual(query.type, constants.serviceTypes.online)) {
+    if (utils.areEqual(query.type, constants.serviceTypes.online)) {
       return 'redirect';
     }
   }
   return undefined;
 }
 
+function getResultsHeading(query, loc) {
+  if (loc) {
+    const location = loc.toUpperCase();
+    if (query.type) {
```

Comment:
Indentation could be reduced by combining the `loc` and `query.type` `if` checks